src/storage/mod.rs

```diff
@@ -858,49 +835,37 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
-                            let cf = match Self::rawkv_cf(&cf) {
-                                Ok(x) => x,
-                                Err(e) => return future::err(e),
-                            };
-                            // no scan_count for this kind of op.
-
-                            let key_len = key.len();
-                            let result = snapshot
-                                .get_cf(cf, &Key::from_encoded(key))
-                                // map storage::engine::Error -> storage::Error
-                                .map_err(Error::from)
-                                .map(|r| {
-                                    if let Some(ref value) = r {
-                                        let mut stats = Statistics::default();
-                                        stats.data.flow_stats.read_keys = 1;
-                                        stats.data.flow_stats.read_bytes = key_len + value.len();
-                                        metrics::tls_collect_read_flow(ctx.get_region_id(), &stats);
-                                        metrics::tls_collect_key_reads(CMD, 1);
-                                    }
-                                    r
-                                });
-                            future::result(result)
-                        })
-                    })
-                    .then(move |r| {
-                        metrics::tls_collect_command_duration(CMD, command_duration.elapsed());
-                        r
-                    })
-            })
-        });
+        let priority = ctx.get_priority();
+        let priority_tag = get_priority_tag(priority);
+
+        let res = self.read_pool.spawn_handle(
+            async move {
+                metrics::tls_collect_command_count(CMD, priority_tag);
+                let command_duration = tikv_util::time::Instant::now_coarse();
+                let snapshot = Self::with_tls_engine(|engine| Self::snapshot(engine, &ctx)).await?;
+                let result = metrics::tls_processing_read_observe_duration(CMD, || {
+                    let cf = Self::rawkv_cf(&cf)?;
+                    // no scan_count for this kind of op.
+
+                    let key_len = key.len();
+                    let r = snapshot.get_cf(cf, &Key::from_encoded(key))?;
+                    if let Some(ref value) = r {
+                        let mut stats = Statistics::default();
+                        stats.data.flow_stats.read_keys = 1;
+                        stats.data.flow_stats.read_bytes = key_len + value.len();
+                        tls_collect_read_flow(ctx.get_region_id(), &stats);
+                        tls_collect_key_reads(CMD, 1);
+                    }
+                    Ok(r)
+                });
+                metrics::tls_collect_command_duration(CMD, command_duration.elapsed());
+                result
+            },
+            priority,
+            thread_rng().next_u64(),
```

Comment:
What does this mean?

Reply:
I think it's just a unique task id.

Reply:
It's used to calculate tasks' duration?

Reply:
Yes. It enables that multiple requests in a single transaction can share the duration statistics. For raw requests, every request needs a unique id.